ocaml/xapi/xapi_host.ml

```diff
@@ -1381,31 +1468,21 @@ let install_server_certificate ~__context ~host ~certificate ~private_key
     Certificates.install_server_certificate ~pem_leaf:certificate
       ~pkcs8_private_key:private_key ~pem_chain
   in
-  let date_of_ptime time = Date.of_float (Ptime.to_float_s time) in
-  let dates_of_ptimes (a, b) = (date_of_ptime a, date_of_ptime b) in
-  let not_before, not_after =
-    dates_of_ptimes (X509.Certificate.validity certificate)
-  in
-  let fingerprint =
-    X509.Certificate.fingerprint Mirage_crypto.Hash.(`SHA256) certificate
-    |> Certificates.pp_hash
-  in
-  let uuid = Uuid.(to_string (make_uuid ())) in
-  let ref = Ref.make () in
   List.iter
     (fun self -> Db.Certificate.destroy ~__context ~self)
     current_server_certs ;
-  Db.Certificate.create ~__context ~ref ~uuid ~host ~not_before ~not_after
-    ~fingerprint ;
+  let (_ : API.ref_Certificate) =
+    add_cert_to_db ~__context ~host ~_type:`host certificate
+  in
   let task = Context.get_task_id __context in
   (* mark task as done *)
   Db.Task.set_progress ~__context ~self:task ~value:1.0 ;
   (* sever all connections done with stunnel *)
   Xapi_mgmt_iface.reconfigure_stunnel ~__context
 
 let reset_server_certificate ~__context ~host =
```

Comment:
Here we've changed `reset_server_certificate` to add the new cert to the database, and remove any old ones. However this means that `emergency_reset_server_certificates` requires a database connection, and therefore won't necessarily work in 'emergency situations' (i.e. slave can't connect to the master, and therefore doesn't have a database connection!)

Perhaps `emergency_reset_server_certificate` should create a new cert, without adding it to the database; then on xapi start up we would need to do a sync of the certs on the file system with the database